test/FastTests/Blittable/BlittableValidationTest.cs

```diff
@@ -0,0 +1,628 @@
+﻿using System;
+using System.IO;
+using System.Text;
+using Raven.Imports.Newtonsoft.Json;
+using Sparrow.Json;
+using Xunit;
+using Raven.Json.Linq;
+using Sparrow.Compression;
+using Sparrow.Json.Parsing;
+
+namespace FastTests.Blittable
+{
+    public class BlittableValidationTest : RavenTestBase
+    {
+        public int Size = 0xbc;
+        public byte[] SimpleValidBlittable = new byte[190]
+        {
+            0x0c, 0x46, 0x69, 0x72, 0x73, 0x74, 0x4e, 0x61, 0x6d, 0x65,
+            0x20, 0x23, 0x30, 0x00, 0x0b, 0x4c, 0x61, 0x73, 0x74, 0x4e,
+            0x61, 0x6d, 0x65, 0x20, 0x23, 0x30, 0x00, 0x00, 0x05, 0x55,
+            0x73, 0x65, 0x72, 0x73, 0x00, 0x1d, 0x54, 0x72, 0x79, 0x6f,
+            0x75, 0x74, 0x73, 0x2e, 0x50, 0x72, 0x6f, 0x67, 0x72, 0x61,
+            0x6d, 0x2b, 0x55, 0x73, 0x65, 0x72, 0x2c, 0x20, 0x54, 0x72,
+            0x79, 0x6f, 0x75, 0x74, 0x73, 0x00, 0x07, 0x75, 0x73, 0x65,
+            0x72, 0x73, 0x2f, 0x30, 0x00, 0x03, 0x09, 0x06, 0x05, 0x28,
+            0x05, 0x05, 0x2f, 0x04, 0x05, 0x04, 0x0a, 0x03, 0x51, 0x55,
+            0x00, 0x05, 0x47, 0x01, 0x05, 0x3a, 0x02, 0x08, 0x09, 0x46,
+            0x69, 0x72, 0x73, 0x74, 0x4e, 0x61, 0x6d, 0x65, 0x00, 0x08,
+            0x4c, 0x61, 0x73, 0x74, 0x4e, 0x61, 0x6d, 0x65, 0x00, 0x04,
+            0x54, 0x61, 0x67, 0x73, 0x00, 0x09, 0x40, 0x6d, 0x65, 0x74,
+            0x61, 0x64, 0x61, 0x74, 0x61, 0x00, 0x11, 0x52, 0x61, 0x76,
+            0x65, 0x6e, 0x2d, 0x45, 0x6e, 0x74, 0x69, 0x74, 0x79, 0x2d,
+            0x4e, 0x61, 0x6d, 0x65, 0x00, 0x0e, 0x52, 0x61, 0x76, 0x65,
+            0x6e, 0x2d, 0x43, 0x6c, 0x72, 0x2d, 0x54, 0x79, 0x70, 0x65,
+            0x00, 0x03, 0x40, 0x69, 0x64, 0x00, 0x10, 0x4e, 0x43, 0x39,
+            0x33, 0x28, 0x15, 0x05, 0x55, 0x01, 0xb0, 0x51, 0x00, 0x00
+        };
+
+        [Fact]
+        public unsafe void EmptyValiedBlittable()
+        {
+            try
+            {
+                using (var pool = new UnmanagedBuffersPool("test"))
+                using (var context = new JsonOperationContext(pool))
+                {
+                    var obj = RavenJObject.FromObject(new Empty());
+                    var objString = obj.ToString(Formatting.None);
+                    var stream = new MemoryStream();
+                    var streamWriter = new StreamWriter(stream);
+                    streamWriter.Write(objString);
+                    streamWriter.Flush();
+                    stream.Position = 0;
+                    var reader = context.Read(stream, "docs/1 ");
+                    reader.BlittableValidation(reader.Size);
+                }
+            }
+            catch (Exception)
+            {
+                Assert.False(true);
```

Comment:
Don't do things like that, you are losing very important information if the test fails. Just let the exception escap.e